libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -410,6 +410,43 @@ bool ExpressionCompiler::visit(TupleExpression const& _tuple)
 bool ExpressionCompiler::visit(UnaryOperation const& _unaryOperation)
 {
 	CompilerContext::LocationSetter locationSetter(m_context, _unaryOperation);
+
+	if (FunctionDefinition const* function = _unaryOperation.annotation().userDefinedFunction)
+	{
+		FunctionType const* functionType = dynamic_cast<FunctionType const*>(
```

Comment:
I think we should assert `function->isFree` here - so we can be sure that we don't have to do virtual function lookup.